maven-core/src/main/java/org/apache/maven/ReactorReader.java

```diff
@@ -139,187 +132,75 @@ private File find(MavenProject project, Artifact artifact) {
-            while (iterator.hasNext()) {
-                Path outputFile = iterator.next();
-
-                if (Files.isDirectory(outputFile)) {
-                    continue;
-                }
-
-                long outputFileLastModified =
-                        Files.getLastModifiedTime(outputFile).toMillis();
-                if (outputFileLastModified > artifactLastModified) {
-                    File alternative = determineBuildOutputDirectoryForArtifact(project, artifact);
-                    if (alternative != null) {
-                        LOGGER.warn(
-                                "File '{}' is more recent than the packaged artifact for '{}'; using '{}' instead",
-                                relativizeOutputFile(outputFile),
-                                project.getArtifactId(),
-                                relativizeOutputFile(alternative.toPath()));
-                    } else {
-                        LOGGER.warn(
-                                "File '{}' is more recent than the packaged artifact for '{}'; "
-                                        + "cannot use the build output directory for this type of artifact",
-                                relativizeOutputFile(outputFile),
-                                project.getArtifactId());
-                    }
-                    return false;
+        for (Artifact artifact : artifacts) {
+            if (artifact.getFile() != null && artifact.getFile().isFile()) {
+                Path target = getArtifactPath(artifact);
+                try {
+                    Files.createDirectories(target.getParent());
+                    Files.copy(artifact.getFile().toPath(), target, StandardCopyOption.REPLACE_EXISTING);
+                } catch (IOException e) {
+                    LOGGER.warn("Error while copying artifact to project local repository", e);
                 }
             }
-
-            return true;
-        } catch (IOException e) {
-            LOGGER.warn(
-                    "An I/O error occurred while checking if the packaged artifact is up-to-date "
-                            + "against the build output directory. "
-                            + "Continuing with the assumption that it is up-to-date.",
-                    e);
-            return true;
         }
     }
 
-    private boolean hasBeenPackagedDuringThisSession(MavenProject project) {
-        return project.hasLifecyclePhase("package")
-                || project.hasLifecyclePhase("install")
-                || project.hasLifecyclePhase("deploy");
+    private Path getArtifactPath(Artifact artifact) {
+        String groupId = artifact.getGroupId();
+        String artifactId = artifact.getArtifactId();
+        String classifier = artifact.getClassifier();
+        String extension = artifact.getExtension();
+        Path root = session.getRequest().getMultiModuleProjectDirectory().toPath();
+        Path repo = root.resolve("target/project-local-repo");
```

Comment:
Will this work as expected on operating systems that have the `\` as the file separator (e.g., Windows)?

Reply:
I think it will be safer as
```
root.resolve("target").resolve("project-local-repo")
```